src/controllers/merchantWebhookQueue.controller.ts

```diff
@@ -0,0 +1,111 @@
+import { Request, Response } from "express";
+import { MerchantWebhookQueueService } from "../services/MerchantWebhookQueue.service";
+
+const merchantWebhookQueueService = new MerchantWebhookQueueService();
+
+
+export class MerchantWebhookQueueController {
+  async getFailedWebhooks(req: Request, res: Response): Promise<any> {
+    try {
+      const { merchantId, limit, offset } = req.query;
+
+      const limitNum = limit ? parseInt(limit as string) : 10;
+      const offsetNum = offset ? parseInt(offset as string) : 0;
+
+      const failedWebhooks =
+        await merchantWebhookQueueService.getFailedWebhookEvents(
+          merchantId as string,
+          limitNum,
+          offsetNum
+        );
```

Comment:
more robust and type-safe
```suggestion
      const merchantId = typeof req.query.merchantId === 'string' ? req.query.merchantId : undefined;
      const limitNum = Math.min(
        parseInt(req.query.limit as string) || 10,
        100
      );
      const offsetNum = Math.max(
        parseInt(req.query.offset as string) || 0,
        0
      );

      const failedWebhooks =
        await merchantWebhookQueueService.getFailedWebhookEvents(
          merchantId as string,
          limitNum,
          offsetNum
        );
```
Please apply to all